### ISSUE TITLE
Fuzz Test Snapshotting Concept

### ISSUE THREAD

**Issue by transmissions11** [gakonst/foundry#1085]
### Component

Forge

### Describe the feature you would like

a cheatcode or config option that lets u specify a couple of specific inputs to snapshot for a fuzz test would be a repeatable and easy way to benchmark fuzz tests without introducing the variance that comes from mean/medians

here's what that could look like:

```solidity
function testAdd(uint a, uint b) public {
    vm.snapshot(1, 2);
    vm.snapshot(10, 60);

    assertEq(a + b, b + a);
}
```

with a modifier it'd be really elegant:

```solidity
modifier snapshot(uint256 a, uint256 b) {
   vm.snapshot(a, b)
}

function testAdd(uint a, uint b) public snapshot(1, 2) snapshot(10, 60) {
     assertEq(a + b, b + a);
}
```

could also use comment annotations but i know we dont rly like those

### Additional context

related to my complaints in https://github.com/gakonst/foundry/issues/1081

**Comment by d-xo** [comment]
related: https://github.com/gakonst/foundry/issues/858

**Comment by transmissions11** [comment]
oh interesting @d-xo yeah seems like a lot of the functionality would be shared between them

**Comment by mds1** [comment]
In practice I think we'd have to do `vm.snapshot(bytes memory)` and ABI encode the params, cause there's basically infinite potential function signatures for fuzz tests

**Comment by transmissions11** [comment]
yup agreed, and can have a console.log style lib as an abstraction on top

**Comment by mds1** [comment]
@transmissions11 Is this worth keeping open? What is the use case for snapshotting only specific values, as opposed to concrete tests for some edge cases you may care about + deterministic seed with fuzz tests for the other cases?

**Comment by zerosnacks** [comment]
Closed by https://github.com/foundry-rs/foundry/pull/7428, see: https://book.getfoundry.sh/forge/fuzz-testing#fuzz-test-fixtures
